### PR TITLE
add documentation on using external modules

### PR DESCRIPTION
**Pull request by boegel**
see https://github.com/hpcugent/easybuild-framework/pull/1265

rendered version @ http://boegel-eb.readthedocs.org/en/external_modules/Using_external_modules.html

### CODE REVIEW

File: docs/Using_external_modules.rst [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+For example, to specify the readily available module ``fftw/3.3.4.2`` as a dependency::
+
+  dependencies = [('fftw/3.3.4.2', EXTERNAL_MODULE)]
+
+For such dependencies, EasyBuild will:
+
+* load the module before initiating the software build and install procedure
+* include a '``module load``' statement in the generated module file (for non-build dependencies)
+
+If the module is not available, EasyBuild will only report the module as being missing. It will *not* search for a
```
Comment (wpoely86) [inline docs/Using_external_modules.rst]: report is a bit vague? EB will stop with an error?

**Comment by wpoely86** [comment]
looks fine. Maybe add a links to a directory with some common metadata files (e.g. Cray)?

File: docs/Using_external_modules.rst (hunk 1 above)
Comment (boegel) [inline docs/Using_external_modules.rst]: thx, fixed

**Comment by boegel** [comment]
@wpoely86: for now, no common metadata files yet, the stuff on Cray is still experimental, we need to figure things better first (e.g., during the upcoming hackathon in Espoo, at a Cray site)

**Comment by pescobar** [comment]
looks fine for me

**Comment by boegel** [comment]
Going in, thanks for the reviews @wpoely86 and @pescobar!

**Pull request merged**